src/ResultWriter/WaveFieldWriter.h

```diff
@@ -91,6 +91,9 @@ class WaveFieldWriter : private async::Module<WaveFieldWriterExecutor, WaveField
 	/** The variable subsampler for the refined mesh */
 	refinement::VariableSubsampler<double>* m_variableSubsampler;
 
+	/** The variable subsampler for the plasticity output */
+	refinement::VariableSubsampler<double>* m_lowVariableSubsampler;
```

Comment:
please use a std::unique_ptr here to ensure that the memory is freed. (Currently its never deleted.)

Reply:
I don't know how to do that.

Reply:
You have to #include <memory>, then you replace
```c++
refinement::VariableSubsampler<double>* m_lowVariableSubsampler;
```
with
```c++
std::unique_ptr<refinement::VariableSubsampler<double>> m_lowVariableSubsampler;
```

and you replace
```c++
m_variableSubsamplerPStrain = new refinement::VariableSubsampler<double>(
			numElems, *tetRefiner, order, WaveFieldWriterExecutor::NUM_PLASTICITY_VARIABLES , numAlignedDOF);
```
with
```c++
m_variableSubsamplerPStrain = std::make_unique<refinement::VariableSubsampler<double>>(
			numElems, *tetRefiner, order, WaveFieldWriterExecutor::NUM_PLASTICITY_VARIABLES , numAlignedDOF);
```

Reply:
thx